### PR TITLE
cmake: enable smoke testing on Windows

### PR DESCRIPTION
**Pull request by apwojcik**
Windows command line (both cmd.exe and PowerShell) does not support providing temporal environmental variables prepended before command invocation. The PR is changing to the ENVIRONMENT property per test command instead.

### CODE REVIEW

**Comment by JehandadKhan** [comment]
@xinlipn  and @reidkwja  Please help review this PR.

**Comment by junliume** [comment]
Ping @xinlipn and @reidkwja before merging

**Review by reidkwja — APPROVED** [review]

**Review by junliume — APPROVED** [review]

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1594,93 +1601,98 @@ set(ARGS_NCHWC_CHWNC_FWD_FP16x8
...
+    COMMAND $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input  400  256 7 7 --weights  256 7 7 1024 --pads_strides_dilations 0 0 1 1 1 1 ${ARGS_NCHWC_CHWNC_FWD_FP16x8}
+    COMMAND $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input  400  256 1 1 --weights  256 1 1 1024 --pads_strides_dilations 0 0 1 1 1 1 ${ARGS_NCHWC_CHWNC_FWD_FP16x8}
 )
 
 add_custom_test(test_regression_half_mi100 SKIP_UNLESS_ALL FLOAT_DISABLED HALF_ENABLED GFX908_ENABLED GFX900_DISABLED GFX906_DISABLED GFX90A_DISABLED
 # Regression test for SWDEV-291202
-COMMAND MIOPEN_FIND_MODE=normal MIOPEN_DEBUG_FIND_ONLY_SOLVER=ConvHipImplicitGemmBwdDataV4R1Xdlops $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input  128  24 14 14 --weights 64  24 5 5 --pads_strides_dilations 2 2 1 1 1 1 --disable-forward --disable-backward-weights
+    COMMAND $<TARGET_FILE:test_conv2d> ${MIOPEN_TEST_FLOAT_ARG} --verbose --input  128  24 14 14 --weights 64  24 5 5 --pads_strides_dilations 2 2 1 1 1 1 --disable-forward --disable-backward-weights
```
Comment (atamazov) [inline test/CMakeLists.txt:1689]: 🐛

**Review by atamazov — COMMENTED** [review]
I'll fix this.

File: test/CMakeLists.txt [hunk 2]
```diff
@@ -2030,265 +2040,239 @@ add_custom_test(smoke_solver_ConvHipImplicitGemmV4R1WrW GFX103X_ENABLED HALF_ENA
...
 
 add_custom_test(test_regression_half_vega SKIP_UNLESS_ALL FLOAT_DISABLED HALF_ENABLED GFX908_DISABLED GFX90A_DISABLED
-# Issue #1956.
-COMMAND	MIOPEN_FIND_MODE=normal MIOPEN_DEBUG_FIND_ONLY_SOLVER='GemmBwdRest' $<TARGET_FILE:test_conv3d> ${TEST_CONV_VERBOSE_B}
-    --cmode conv --pmode default --group-count 1 --batch_size 2 --input_channels 64 --output_channels 32 --spatial_dim_elements 128 128 128
-    --filter_dims 3 3 3 --pads_strides_dilations 1 1 1 1 1 1 1 1 1 --trans_output_pads 0 0 0
-    --in_layout NCDHW --fil_layout NCDHW --out_layout NCDHW ${MIOPEN_TEST_FLAGS_ARGS}
+    # Issue #1956.
+    ENVIRONMENT MIOPEN_FIND_MODE=normal MIOPEN_DEBUG_FIND_ONLY_SOLVER='GemmBwdRest'
+    COMMAND $<TARGET_FILE:test_conv3d> ${TEST_CONV_VERBOSE_B} --cmode conv --pmode default --group-count 1 --batch_size 2 --input_channels 64 --output_channels 32 --spatial_dim_elements 128 128 128 --filter_dims 3 3 3 --pads_strides_dilations 1 1 1 1 1 1 1 1 1 --trans_output_pads 0 0 0 --in_layout NCDHW --fil_layout NCDHW --out_layout NCDHW ${MIOPEN_TEST_FLAGS_ARGS}
```
Comment (atamazov) [inline test/CMakeLists.txt:2275]: @apwojcik Please avoid whitespace-only (formatting-only) changes in the PRs that change behavior of the library. Please submit whitespace/formatting-only changes as separate PRs and mark them "[NFC]". Thanks.

**Comment by atamazov** [comment]
@xinlipn @JehandadKhan FYI from now on, we can't mix different env settings in the same custom test.

**Pull request merged**